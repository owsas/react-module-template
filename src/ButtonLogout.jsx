import React from 'react';
import PropTypes from 'prop-types';
import PowerOff from 'react-icons/lib/fa/power-off';

const ButtonLogout = props => (
  <button className="btn btn-danger" onClick={props.onClick}>
    <span style={{ marginRight: 5 }}>
      <PowerOff />
    </span>
    Cerrar sesi√≥n
  </button>
);

ButtonLogout.defaultProps = {
  onClick: undefined,
};

ButtonLogout.propTypes = {
  onClick: PropTypes.func,
};

export default ButtonLogout;
