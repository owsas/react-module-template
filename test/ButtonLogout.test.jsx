import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ButtonLogout from '../src/ButtonLogout';

test('should match snapshot', () => {
  const m = shallow(<ButtonLogout
    onClick={() => {}}
  />);

  expect(toJSON(m)).toMatchSnapshot();
});
