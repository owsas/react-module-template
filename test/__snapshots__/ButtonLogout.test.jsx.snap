// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<button
  className="btn btn-danger"
  onClick={[Function]}
>
  <span
    style={
      Object {
        "marginRight": 5,
      }
    }
  >
    <FaPowerOff />
  </span>
  Cerrar sesi√≥n
</button>
`;
